docs/examples/DataSet/DataSet-class-walkthrough.ipynb

```diff
@@ -0,0 +1,1327 @@
+      "text/plain": [
+       "<Figure size 432x288 with 2 Axes>"
+      ]
+     },
+     "metadata": {
+      "needs_background": "light"
+     },
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "plot_dataset(dataset)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Note on immutability of DataSet\n",
+    "\n",
+    "Users are NOT expected to use `DataSet`s methods that modify its content, for example, `add_result`. Only `Measurement` and `DataSaver` objects use those methods. This is due to the phylosophy of the `DataSet` - once created within a `Measurement` context, it should not change so that the user don't accidentally modify/remove the important measured data. In the future versions of QCoDeS the design of the `DataSet` object will be improved to make the described phylosophy clear."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## DataSet indentity\n",
+    "\n",
+    "Before we dive into what's in the `DataSet`, let's briefly note how a `DataSet` is identified.\n",
+    "\n",
+    "TL;DR:\n",
+    "> * Both `experiment name` and `sample name` are great means for identifying a dataset, especially when coupled with the `captured_run_id`.\n",
+    "> * `guid` is the only true globally unique identifier for datasets."
```

Comment:
I am not clear on what "globally true" means. Can I have locally true but globally not unique identifiers?

Reply:
haha)) it's not "globally true", it is "true" (slang) and "globally unique". i'll rewrite to avoid confusions here

Reply:
Oh OK, but this was not clear :))